metric-publishers/cloudwatch-metric-publisher/src/test/java/software/amazon/awssdk/metrics/publishers/cloudwatch/internal/transform/MetricCollectionAggregatorTest.java

```diff
@@ -0,0 +1,403 @@
+                                                                               .sum(14.0)
+                                                                               .sampleCount(5.0)
+                                                                               .build());
+            });
+        });
+    }
+
+    @Test
+    public void metricSummariesAreCorrectWithValuesInDifferentCollector() {
+        MetricCollectionAggregator aggregator = defaultAggregator();
+
+        MetricCollector collector = collector();
+        collector.reportMetric(CoreMetric.SERVICE_ID, "ServiceId");
+        collector.reportMetric(HttpMetric.MAX_CONCURRENCY, 2);
+        aggregator.addCollection(collectToFixedTime(collector));
+
+        collector = collector();
+        collector.reportMetric(CoreMetric.SERVICE_ID, "ServiceId");
+        collector.reportMetric(HttpMetric.MAX_CONCURRENCY, 1);
+        aggregator.addCollection(collectToFixedTime(collector));
+
+        collector = collector();
+        collector.reportMetric(CoreMetric.SERVICE_ID, "ServiceId");
+        collector.reportMetric(HttpMetric.MAX_CONCURRENCY, 4);
+        aggregator.addCollection(collectToFixedTime(collector));
+
+        collector = collector();
+        collector.reportMetric(CoreMetric.SERVICE_ID, "ServiceId");
+        collector.reportMetric(HttpMetric.MAX_CONCURRENCY, 4);
+        aggregator.addCollection(collectToFixedTime(collector));
+
+        collector = collector();
+        collector.reportMetric(CoreMetric.SERVICE_ID, "ServiceId");
+        collector.reportMetric(HttpMetric.MAX_CONCURRENCY, 3);
+        aggregator.addCollection(collectToFixedTime(collector));
+
+        assertThat(aggregator.getRequests()).hasOnlyOneElementSatisfying(request -> {
+            assertThat(request.namespace()).isEqualTo(DEFAULT_NAMESPACE);
+            assertThat(request.metricData()).hasOnlyOneElementSatisfying(metricData -> {
+                assertThat(metricData.dimensions()).hasOnlyOneElementSatisfying(dimension -> {
+                    assertThat(dimension.name()).isEqualTo(CoreMetric.SERVICE_ID.name());
+                    assertThat(dimension.value()).isEqualTo("ServiceId");
+                });
+                assertThat(metricData.values()).isEmpty();
+                assertThat(metricData.counts()).isEmpty();
+                assertThat(metricData.statisticValues()).isEqualTo(StatisticSet.builder()
+                                                                               .minimum(1.0)
+                                                                               .maximum(4.0)
+                                                                               .sum(14.0)
+                                                                               .sampleCount(5.0)
+                                                                               .build());
+            });
+        });
+    }
+
+    @Test
+    public void detailedMetricsAreCorrect() {
```

Comment:
Can we have tests that probe the boundary conditions to do with upload limits, like what happens if the publish is on the boundary, just over and just under kinda thing?

Reply:
How is that different than `MetricCollectionAggregatorTest#maximumRequestsIsHonored` and `MetricCollectionAggregatorTest#maximumMetricValuesIsHonored`?

Reply:
Oh yeah true, do you think we should be testing the aggregation/values published in those tests as well as the counts?

Reply:
I'll try it out. I don't think that's covered elsewhere.